node/src/blockchain/blockchain_interface/blockchain_interface_web3/lower_level_interface_web3.rs

```diff
@@ -1,395 +1,762 @@
+                    .unwrap()
+                    .address(),
             )
-            .unwrap();
+            .wait();
+
+        match result {
+            Err(BlockchainError::QueryFailed(msg)) if msg.contains("invalid hex character: Q") => {
+                ()
+            }
+            x => panic!("Expected complaint about hex character, but got {:?}", x),
+        };
+    }
+
+    #[test]
+    fn get_gas_price_works() {
+        let port = find_free_port();
+        let _blockchain_client_server = MBCSBuilder::new(port)
+            .response("0x01".to_string(), 1)
+            .start();
+        let subject = make_blockchain_interface_web3(Some(port));
+
+        let result = subject.lower_interface().get_gas_price().wait().unwrap();
+
+        assert_eq!(result, 1.into());
+    }
+
+    #[test]
+    fn get_gas_price_returns_error() {
+        let port = find_free_port();
+        let _blockchain_client_server = MBCSBuilder::new(port).start();
+        let subject = make_blockchain_interface_web3(Some(port));
+
+        let error = subject
+            .lower_interface()
+            .get_gas_price()
+            .wait()
+            .unwrap_err();
 
-        assert_eq!(result, U256::from(0xDEADBEEF_u64));
-        let requests = test_server.requests_so_far();
-        let bodies: Vec<Value> = requests
-            .into_iter()
-            .map(|request| serde_json::from_slice(&request.body()).unwrap())
-            .collect();
-        assert_eq!(bodies[0]["method"].to_string(), "\"eth_getBalance\"",);
         assert_eq!(
-            bodies[0]["params"][0].to_string(),
-            "\"0x3f69f9efd4f2592fd70be8c32ecd9dce71c472fc\"",
+            error,
+            QueryFailed("Transport error: Error(IncompleteMessage)".to_string())
         );
-        assert_eq!(bodies.len(), 1)
     }
 
     #[test]
-    #[should_panic(expected = "No address for an uninitialized wallet!")]
-    fn low_interface_web3_get_transaction_fee_balance_returns_err_for_an_invalid_wallet() {
-        let port = 8545;
-        let (_event_loop_handle, transport) = Http::with_max_parallel(
-            &format!("http://{}:{}", &Ipv4Addr::LOCALHOST.to_string(), port),
-            REQUESTS_IN_PARALLEL,
-        )
-        .unwrap();
-        let chain = TEST_DEFAULT_CHAIN;
-        let subject = make_subject(transport, chain);
+    fn get_block_number_works() {
+        let port = find_free_port();
+        let _blockchain_client_server = MBCSBuilder::new(port)
+            .response("0x23".to_string(), 1)
+            .start();
+        let subject = make_blockchain_interface_web3(Some(port));
 
-        let result = subject.get_transaction_fee_balance(&Wallet::new("0x_invalid_wallet_address"));
+        let result = subject.lower_interface().get_block_number().wait();
 
-        assert_eq!(result, Err(BlockchainError::InvalidAddress));
+        assert_eq!(result, Ok(35.into()));
```

Comment:
Also here it would be appreciated if you could put it as hex.  Maybe please alter this number so that you don't assert on the same one all the time. 

Again, don't dare to forget about the asserting on the request's json and the method's specification (the web3 opcode).